docs/releases.rst

```diff
@@ -1,6 +1,333 @@
+
+For more information refer to the `Mailu RESTful API` page.
+
+Header authentication support (use external identity providers)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is now possible to use different authentication systems (such as keycloak, authentik, vouch-proxy) to handle the authentication of Mailu users.
+This can be used to enable Single Sign On from other IDentity Providers via protocols such as OIDC or SAML2.
+
+For more information see `Header authentication using an external proxy` in the configuration reference.
+
+Better anti-spoofing protection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously Mailu would reject emails where an attacker spoofs the envelope-From. Now Mailu also checks the header-From for any hosted domain.
+It won't let any email which pretends to be for any of the local domains through unless they pass DMARC.
+
+Implement a password policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In line with security best practices, we have introduced password policy.
+
+Passwords now need to:
+- be at least 8 characters long
+- not be listed on `HaveIBeenPwned <https://haveibeenpwned.com/Passwords>`_
+
+This mirrors word-for-word the advice from `NIST Special Publication 800-63B <https://pages.nist.gov/800-63-3/sp800-63b.html#5111-memorized-secret-authenticators>`_.
+
+Significant improvements to the Rate-limiter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now the rate limiter will only take distinct attempts into account. We have two different types of checks:
+- to prevent crendential bruteforce (an attacker trying to guess a password), we limit the maximal amount of attempts an attacker has for a given account (from any IP address)
+- to prevent password spraying (an attacker trying the same common password on all accounts he can enumerate), we limit the maximal number of non-existing accounts an attacker can attempt to authenticate against from a given network subnet.
+
+We have also implemented state-of-the-art features such as `Device Cookies <https://owasp.org/www-community/Slow_Down_Online_Guessing_Attacks_with_Device_Cookies>`_ and IP-whitelisting post-authentication to ensure we don't lock genuine users out.
+
+Rate-limiters have a bad name because they are often misunderstood. If you used Mailu's rate-limiter in the past and had a bad experience please consider giving it another try after upgrading.
+
+Remember the login URL
+^^^^^^^^^^^^^^^^^^^^^^
+
+Mailu will now remember which URL was requested and redirect you to it post-authentication.
+
+This functionality can be used by visiting a "deep" URL E.g.
+
+- https://test.mailu.io/admin
+- https://test.mailu.io/webmail
+
+This results in a login page with a single login button. To access the normal login page, visit the root url.
+
+- https://test.mailu.io
+
+Users who only use the /admin endpoint can now bookmark https://test.mailu.io/admin. When logging in, it is possible to use the `Enter` key again.
```

Comment:
"When logging in, it is possible to use the `Enter` key again" => not really clear what that means